client/src/components/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect } from 'react';
+import { useAuth } from '../../context/useAuthContext';
+import { useSocket } from '../../context/useSocketContext';
+import CircularProgress from '@mui/material/CircularProgress';
+import { Route, Redirect, RouteProps, RouteComponentProps } from 'react-router-dom';
+
+const ProtectedRoute: React.FC<RouteProps> = ({ component, render, ...routeProps }) => {
+  const { loggedInUser } = useAuth();
+  const { initSocket } = useSocket();
+
+  useEffect(() => {
```

Comment:
You can remove this `useEffect` as we don't need to initialize the socket connection here. Ideally, we would want to do it in the `useSocketContext`

Reply:
Okay, I'll remove it